exonum/src/runtime/rust/mod.rs

```diff
@@ -0,0 +1,174 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{
+    cell::RefCell,
+    collections::{HashMap, HashSet},
+};
+
+#[macro_use]
+pub mod service;
+
+#[cfg(test)]
+pub mod tests;
+
+use super::{
+    error::{DeployError, ExecutionError, InitError},
+    ArtifactSpec, CallInfo, DeployStatus, EnvContext, InstanceInitData, RuntimeEnvironment,
+    ServiceInstanceId,
+};
+
+use crate::crypto::{Hash, PublicKey};
+use crate::storage::Fork;
+
+use self::service::SystemService;
+
+#[derive(Debug, Default)]
+struct RustRuntime {
+    inner: RefCell<RustRuntimeInner>,
```

Comment:
Yes, but here start_deploy does not call check_deploy_status so it would not happen and this code would not compile if we will try to run in multiple threads. Do you propose to have lock here? But then, I think, it may be an issue if we will have to take a lock every time we execute tx.

Reply:
I propose to keep this thing in mind :) Later we'll test this carefully and find a proper solution. I guess we don't have to solve the problem that doesn't yet exists :)
Maybe just mark it with `TODO`?